src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisConfigDrawer.tsx

```diff
@@ -0,0 +1,191 @@
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import {
+  Drawer,
+  DrawerActions,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerContentBody,
+  DrawerHead,
+  DrawerPanelContent,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  DropdownToggleAction,
+  Level,
+  LevelItem,
+} from '@patternfly/react-core';
+import { CogIcon, PlusCircleIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+import { first } from 'rxjs';
+import { AutomatedAnalysisConfigForm } from './AutomatedAnalysisConfigForm';
+
+interface AutomatedAnalysisConfigDrawerProps {
+  drawerContent: React.ReactNode;
+  onCreate: () => void;
+}
+
+export const AutomatedAnalysisConfigDrawer: React.FunctionComponent<AutomatedAnalysisConfigDrawerProps> = (props) => {
+  const context = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+
+  const [isExpanded, setIsExpanded] = React.useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+  const drawerRef = React.useRef<HTMLDivElement>(null);
+
+  const onToggle = React.useCallback(
+    (selected) => {
+      setIsDropdownOpen(selected);
+    },
+    [setIsDropdownOpen]
+  );
+
+  const onDropdownFocus = () => {
+    const element = document.getElementById('automated-analysis-recording-config-dropdown');
+    element?.focus();
+  };
+
+  const handleCreateRecording = React.useCallback(
+    (recordingAttributes: RecordingAttributes) => {
+      addSubscription(
+        context.api
+          .createRecording(recordingAttributes)
+          .pipe(first())
+          .subscribe({
+            next: (resp) => {
+              props.onCreate();
+            },
+            error: (err) => {},
+          })
+      );
+    },
+    [addSubscription, context.api, props.onCreate]
+  );
+
+  const onDefaultRecordingStart = React.useCallback(() => {
+    const config = context.settings.automatedAnalysisRecordingConfig();
+    const attributes = automatedAnalysisConfigToRecordingAttributes(config);
+    handleCreateRecording(attributes);
+  }, [context.settings, context.settings.automatedAnalysisRecordingConfig, handleCreateRecording]);
+
+  const onDropdownSelect = React.useCallback(() => {
+    setIsDropdownOpen(false);
+    onDropdownFocus();
+  }, [setIsDropdownOpen]);
+
+  const onExpand = () => {
+    drawerRef.current && drawerRef.current.focus();
+  };
+
+  const onClick = React.useCallback(() => {
+    setIsExpanded(!isExpanded);
+  }, [setIsExpanded, isExpanded]);
+
+  const onCloseClick = React.useCallback(() => {
+    setIsExpanded(false);
+  }, [setIsExpanded]);
+
+  const panelContent = React.useMemo(() => {
+    return (
+      <DrawerPanelContent>
+        <DrawerHead>
+          <span tabIndex={isExpanded ? 0 : -1} ref={drawerRef}>
+            <AutomatedAnalysisConfigForm onCreate={props.onCreate} isSettingsForm={false} />
```

Comment:
This should be in `DrawerPanelBody`. I think you can check `RecordingLabelsPanel.tsx` for references :D

Reply:
The `DrawerHead` should only contain `DrawerActions`, which is the close button :D